app/src/lib/stores/git-store.ts

```diff
@@ -1516,22 +1517,25 @@ export class GitStore extends BaseStore {
     await queueWorkHigh(files, async file => {
       const foundSubmodule = submodules.some(s => s.path === file.path)
 
-      if (
-        file.status.kind !== AppFileStatusKind.Deleted &&
-        !foundSubmodule &&
-        moveToTrash
-      ) {
-        // N.B. moveItemToTrash can take a fair bit of time which is why we're
-        // running it inside this work queue that spreads out the calls across
-        // as many animation frames as it needs to.
-        try {
-          await this.shell.moveItemToTrash(
-            Path.resolve(this.repository.path, file.path)
-          )
-        } catch (e) {
-          if (askForConfirmationOnDiscardChangesPermanently) {
-            throw new DiscardChangesError(e, this.repository, files)
+      if (file.status.kind !== AppFileStatusKind.Deleted && !foundSubmodule) {
+        if (moveToTrash) {
+          // N.B. moveItemToTrash can take a fair bit of time which is why we're
+          // running it inside this work queue that spreads out the calls across
+          // as many animation frames as it needs to.
+          try {
+            await this.shell.moveItemToTrash(
+              Path.resolve(this.repository.path, file.path)
+            )
+          } catch (e) {
+            if (askForConfirmationOnDiscardChangesPermanently) {
+              throw new DiscardChangesError(e, this.repository, files)
+            }
           }
+        } else if (file.status.kind === AppFileStatusKind.Untracked) {
+          // Removing untracked files as they will not be reset since they are
+          // no detected as changes in the index. We need to remove them
+          // manually.
+          rm(Path.join(repoPath, file.path))
```

Comment:
We need to await this, and (I imagine) also handle if the file can't be deleted (locked, in use, etc)

Reply:
Added the await -> If this would fail there is a catch block on the in the `_discardChanges` at the app-store level that would show a user an error in that case.